orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -1409,6 +1409,10 @@ private void callBackupRestore(GatewayConfig primaryGateway, Set<String> target,
             saltJobRunBootstrapRunner.call();
         } catch (Exception e) {
             LOGGER.error("Error occurred during database backup/restore", e);
+            if (e.getMessage().contains("jobs.lookup_jid returns an empty response")) {
```

Comment:
please don't match internal exceptions by the message. either catch `SaltExecutionWentWrongException` or create a new one

Reply:
@lacikaaa Thanks a lot for the suggestions! Updated it to catch using the exception. Please let me know if it looks good. Thank you!

Reply:
Let me think about how to do that. I just realize here will not catch the `SaltExecutionWentWrongException `. Will let you know when it's updated!

Reply:
@lacikaaa May I ask "create a new one" - does this mean I should create a new kind of exception? It seems if I need to catch `SaltExecutionWentWrongException ` there I would need to make changes to many bootstrap runner core functions which I would try to avoid.

Reply:
```suggestion
 catch (SaltExecutionWentWrongException saltExecutionWentWrongException) {
            String errorMessage = saltExecutionWentWrongException.getMessage() + " This might be caused by there's no space left on the device.";
            throw new CloudbreakOrchestratorFailedException(errorMessage, saltExecutionWentWrongException);
        } catch (Exception e) {
            LOGGER.error("Error occurred during database backup/restore", e);
            throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
        }
```

Reply:
Thank you @kkalvagadda1 ! We will not be able to catch `SaltExecutionWentWrongException ` there since the this function `callBackupRestore()` will call `saltJobRunBootstrapRunner.call()` which will call `doCall()` and then `checkResult` (this only throws `CloudbreakOrchestratorTimeoutException` or `CloudbreakOrchestratorFailedException`) in `OrchestratorBootstrapRunner.java`. In `doCall()` it will call `call()` in `SaltJobIdTracker.java`... after doing some calls it will finally call `applyStateJidInfo` where we throw `SaltExecutionWentWrongException`. So we have already lost `SaltExecutionWentWrongException` long time ago along the process. What I can do is I can catch `CloudbreakOrchestratorException` in there, but we still need to check the error message to make sure this is the error we want to add in the string `This might be caused by there's no space left on the device.` Please let me know if this would work for you. CC @lacikaaa  Thank you so much!!

`catch (CloudbreakOrchestratorException e) {
            LOGGER.error("Error occurred during database backup/restore", e);
            if (e.getMessage().contains("jobs.lookup_jid returns an empty response")) {
                String errorMessage = e.getMessage() + " This might be caused by there's no space left on the device.";
                throw new CloudbreakOrchestratorFailedException(errorMessage, e);
            } else {
                throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
            }
        }
        catch (Exception e) {
            LOGGER.error("Error occurred during database backup/restore", e);
            throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
        }`

Reply:
Thank you @kkalvagadda1 ! We will not be able to catch `SaltExecutionWentWrongException ` there since the this function `callBackupRestore()` will call `saltJobRunBootstrapRunner.call()` which will call `doCall()` and then `checkResult` (this only throws `CloudbreakOrchestratorTimeoutException` or `CloudbreakOrchestratorFailedException`) in `OrchestratorBootstrapRunner.java`. In `doCall()` it will call `call()` in `SaltJobIdTracker.java`... after doing some calls it will finally call `applyStateJidInfo` where we throw `SaltExecutionWentWrongException`. So we have already lost `SaltExecutionWentWrongException` long time ago along the process. What I can do is I can catch `CloudbreakOrchestratorException` in there, but we still need to check the error message to make sure this is the error we want to add in the string `This might be caused by there's no space left on the device.` Please let me know if this would work for you. CC @lacikaaa  Thank you so much!!

`catch (CloudbreakOrchestratorException e) {
            LOGGER.error("Error occurred during database backup/restore", e);
            if (e.getMessage().contains("jobs.lookup_jid returns an empty response")) {
                String errorMessage = e.getMessage() + " This might be caused by there's no space left on the device.";
                throw new CloudbreakOrchestratorFailedException(errorMessage, e);
            } else {
                throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
            }
        }
        catch (Exception e) {
            LOGGER.error("Error occurred during database backup/restore", e);
            throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
        }`

Reply:
Hi @lacikaaa , as per the conversation with Kalyan, I updated this to catch `CloudbreakOrchestratorException ` but we still need to match the error message though. Please let me know if the updated code looks good to you. Thank you so much!!

Reply:
No, it doesn't. I still can't see why don't you just create a new exception which extends eg: `CloudbreakOrchestratorFailedException`
This exception doesn't have to be a runtimeexception anyway.

Reply:
Hi @lacikaaa , thank you for the explanation over the Slack. I updated the code by making changes to some salt methods. Please let me know if it looks good. Thank you!!

Reply:
Hi @lacikaaa , sorry for the back and forth. I updated the code and now we are not throwing `CloudbreakOrchestratorFailedException ` when the jid info is empty, instead, we set that case to be in progress (`CloudbreakOrchestratorInprogressException `), because during our test we find out if that is set to be failed, `saltJobIdTracker` will trigger a new jid job, but the old jid job was still running by salt and the new jid job will never get started, which will finally still mark this job as finished & successful. Please let me know if the updated code looks good to you. Tests are updated in the description. Thank you so much!!

Reply:
Hi @lacikaaa Would like to follow up on this one. Thanks a lot, again!=)